internals/cli/cmd_enter.go

```diff
@@ -123,7 +129,7 @@ func (cmd *cmdEnter) Execute(args []string) error {
 		extraArgs []string
 	)
 
-	parser := Parser(cmd.client)
+	parser := Parser(cmd.client, cmd.runOptions)
```

Comment:
Is this the correct semantics for the `enter` command?

At least the unit tests fail and I'm not sure what the real issue is:

```
... Panic: internal error: daemon stopped early (PC=0x46F442)

/opt/hostedtoolcache/go/1.20.14/x64/src/runtime/panic.go:890
  in gopanic
export_test.go:102
  in PebbleMain.func2
/opt/hostedtoolcache/go/1.20.14/x64/src/runtime/panic.go:890
  in gopanic
cli.go:285
  in Run.func1
/opt/hostedtoolcache/go/1.20.14/x64/src/runtime/panic.go:890
  in gopanic
cmd_enter.go:187
  in cmdEnter.Execute
/home/runner/go/pkg/mod/github.com/canonical/go-flags@v0.0.0-20230403090104-105d09a091b8/parser.go:335
  in Parser.ParseArgs
/home/runner/go/pkg/mod/github.com/canonical/go-flags@v0.0.0-20230403090104-105d09a091b8/parser.go:191
  in Parser.Parse
cli.go:309
  in Run
export_test.go:26
  in RunMain
export_test.go:106
  in PebbleMain
cmd_enter_test.go:158
  in PebbleSuite.TestEnterExecListDir
/opt/hostedtoolcache/go/1.20.14/x64/src/reflect/value.go:370
  in Value.Call
/opt/hostedtoolcache/go/1.20.14/x64/src/runtime/asm_amd64.s:1598
  in goexit
PANIC: cmd_enter_test.go:146: PebbleSuite.TestEnterExecListDir
```

Reply:
If you merge the latest master into this branch to get the [reaper.Stop err fix](https://github.com/canonical/pebble/pull/390), you'll get a better error message, which is:

```
cannot run daemon: directory "" must be absolute
```

It's happening because somehow the PebbleDir isn't being wired up, so it's empty string when it gets to overlord.New. I'm not sure exactly why it's not being wired up.

Reply:
The `cmdRun` created in `cmd_enter.go` was not getting `pebbleDir` initialized. Pushed a fix.